### PR TITLE
chore: Enable strict mode in qualtrics library and add additional type-safety #10924

### LINKED ISSUE #10924: Enable strict mode in qualtrics library and add additional type-safety
For any new feature-libs, we want to enable strict mode

### PR DESCRIPTION
**Pull request by bgambocjaviniar**
closes GH-10924

### CODE REVIEW

File: feature-libs/qualtrics/components/qualtrics-loader/qualtrics-loader.service.ts [hunk 1]
```diff
@@ -50,16 +57,19 @@ export class QualtricsLoaderService implements OnDestroy {
...
-    ),
-    filter((api) => Boolean(api)),
-    tap((qsi) => (this.qsiApi = qsi))
+    map((dataLoaded: boolean) => (dataLoaded ? this.window?.QSI : EMPTY)),
+    filter((api: any) => Boolean(api)),
+    tap((qsi: any) => (this.qsiApi = qsi))
   );
 
+  get window(): QualtricsWindow | undefined {
+    return this.winRef?.nativeWindow;
```
Comment (znikola) [inline feature-libs/qualtrics/components/qualtrics-loader/qualtrics-loader.service.ts]: `winRef` is always present, no? So no need to do `winRef?`

File: feature-libs/qualtrics/components/qualtrics-loader/qualtrics-loader.service.ts [hunk 2]
```diff
@@ -50,16 +57,19 @@ export class QualtricsLoaderService implements OnDestroy {
    */
   protected qsi$: Observable<any> = this.qsiLoaded$.pipe(
     switchMap(() => this.isDataLoaded()),
-    map((dataLoaded) =>
-      dataLoaded ? this.winRef?.nativeWindow['QSI'] : EMPTY
-    ),
-    filter((api) => Boolean(api)),
-    tap((qsi) => (this.qsiApi = qsi))
+    map((dataLoaded: boolean) => (dataLoaded ? this.window?.QSI : EMPTY)),
+    filter((api: any) => Boolean(api)),
```
Comment (znikola) [inline feature-libs/qualtrics/components/qualtrics-loader/qualtrics-loader.service.ts]: Let's be consistent with the `tap` below, and rename `api` to `qsi`

File: feature-libs/qualtrics/components/qualtrics-loader/qualtrics-loader.service.ts [hunk 3]
```diff
@@ -38,8 +45,8 @@ export class QualtricsLoaderService implements OnDestroy {
   /**
    * QSI load event that happens when the QSI JS file is loaded.
    */
-  private qsiLoaded$: Observable<any> = this.winRef?.nativeWindow
-    ? fromEvent(this.winRef.nativeWindow, QUALTRICS_EVENT_NAME)
+  private qsiLoaded$: Observable<any> = isPlatformBrowser(this.platformId)
+    ? fromEvent(this.window as QualtricsWindow, QUALTRICS_EVENT_NAME)
```
Comment (znikola) [inline feature-libs/qualtrics/components/qualtrics-loader/qualtrics-loader.service.ts]: You don't need to cast it to `QualtricsWindow` here, as the `this.window` already returns it.
You most likely have to check if the `this.window` exists, and you can do it like this: `isPlatformBrowser(this.platformId) && this.window`

File: feature-libs/qualtrics/components/qualtrics-loader/qualtrics-loader.service.ts [hunk 4]
```diff
@@ -50,16 +57,19 @@ export class QualtricsLoaderService implements OnDestroy {
...
 
+  get window(): QualtricsWindow | undefined {
+    return this.winRef?.nativeWindow;
+  }
+
   constructor(
     protected winRef: WindowRef,
-    protected rendererFactory: RendererFactory2
+    protected rendererFactory: RendererFactory2,
+    @Inject(PLATFORM_ID) protected platformId: any
```
Comment (znikola) [inline feature-libs/qualtrics/components/qualtrics-loader/qualtrics-loader.service.ts:73]: A breaking change? 🤔

**Review by znikola — CHANGES_REQUESTED** [review]

File: feature-libs/qualtrics/components/qualtrics-loader/qualtrics-loader.service.ts (hunk 4 above)
Comment (bgambocjaviniar) [reply feature-libs/qualtrics/components/qualtrics-loader/qualtrics-loader.service.ts:73]: As discussed, it is most likely not due to the fact that it still exist in the storefrontlib, where the feature used to live.

Comment (bgambocjaviniar) [reply feature-libs/qualtrics/components/qualtrics-loader/qualtrics-loader.service.ts:73]: Migration plan was to just put deprecation comments in the storefrontlib and move everything to the feature-lib, so I believe it is not a breaking change, but a 'new' feature

**Review by znikola — APPROVED** [review]

**Pull request merged**